compiler/erg_parser/lex.rs

```diff
@@ -688,6 +688,96 @@ impl Lexer /*<'a>*/ {
         ))
     }
 
+    fn lex_multi_line_str(&mut self) -> LexResult<Token> {
+        let mut s = "\"\"\"".to_string();
+        while let Some(c) = self.peek_cur_ch() {
+            match c {
+                '"' => {
+                    s.push(self.consume().unwrap());
+                    let next_c = self.peek_cur_ch();
+                    let aft_next_c = self.peek_next_ch();
+                    if next_c.is_none() || aft_next_c.is_none() {
+                        let token = self.emit_token(Illegal, &s);
+                        return Err(LexError::syntax_error(
+                            0,
+                            token.loc(),
+                            switch_lang!(
+                                "japanese" => "文字列が\"\"\"によって閉じられていません",
+                                "simplified_chinese" => "字符串没有被\"\"\"关闭",
+                                "traditional_chinese" => "字符串没有被\"\"\"关闭",
+                                "english" => "the string is not closed by \"\"\"",
+                            ),
+                            None,
+                        ));
+                    }
+                    let next_c = self.consume().unwrap();
+                    let aft_next_c = self.consume().unwrap();
+                    if next_c == '"' && aft_next_c == '"' {
+                        let token = self.emit_token(StrLit, &s);
+                        return Ok(token);
+                    }
+                }
+                _ => {
+                    let c = self.consume().unwrap();
+                    if c == '\\' {
+                        let next_c = self.consume().unwrap();
+                        match next_c {
+                            '0' => s.push('\0'),
+                            'r' => s.push('\r'),
+                            '\'' => {
+                                s.push('\'');
+                                if self.peek_next_ch().is_some()
+                                    && self.peek_next_ch().unwrap() == '\n'
+                                {
+                                    continue; // Escaping a line break if only '\' comes at the end
+                                }
+                            }
+                            '"' => s.push('"'),
+                            't' => s.push_str("    "), // tab is invalid, so changed into 4 whitespace
+                            '\\' => s.push('\\'),
+                            'n' => s.push('\n'),
```

Comment:
Do I need to add "self.lineno..." below here as well?

Reply:
Yes, I am. This is required for the location of the token.

Reply:
I will add this and a test for tokenization.